### PR TITLE
Mongo32 support

### PR DESCRIPTION
**Pull request by kwojtaszek**
Fixed truncate (system.namespaces are depracated in favor of listCollections command) and handling constraint matching differently to have tests passing properly

### CODE REVIEW

File: lib/mongo_ecto.ex [hunk 1]
```diff
@@ -769,6 +775,7 @@ defmodule Mongo.Ecto do
     end)
   end
 
+
```
Comment (michalmuskala) [inline lib/mongo_ecto.ex]: :scissors:

File: lib/mongo_ecto/connection.ex [hunk 2]
```diff
@@ -124,22 +124,29 @@ defmodule Mongo.Ecto.Connection do
       fun.()
     rescue
       e in Mongo.Error ->
-        stacktrace = System.stacktrace
         case e do
           %Mongo.Error{code: 11000, message: msg} ->
             {:invalid, constraint(msg)}
           other ->
+            stacktrace = System.stacktrace
```
Comment (michalmuskala) [inline lib/mongo_ecto/connection.ex]: Why is this moved? It's better to acquire the stacktrace for reraising at the beginning of error handling.

File: lib/mongo_ecto/connection.ex [hunk 3]
```diff
@@ -124,22 +124,29 @@ defmodule Mongo.Ecto.Connection do
...
+      fn(msg) -> String.split(msg, ["index: ", " dup "]) end
+    end
+
+    case matchf.(msg) do
       [_, name, _] ->
         [unique: String.strip(name)]
       _other ->
         []
     end
+
```
Comment (michalmuskala) [inline lib/mongo_ecto/connection.ex]: :scissors:

File: lib/mongo_ecto.ex [hunk 4]
```diff
@@ -758,7 +758,13 @@ defmodule Mongo.Ecto do
   @list_collections_query ["$and": [[name: ["$not": special_regex]],
                                     [name: ["$not": migration_regex]]]]
 
-  defp list_collections(repo, opts) do
+  def list_collections(_ = "3.2", repo, opts) do
+    colls = command(repo, %{"listCollections": 1}, opts)
+
+    colls["cursor"]["firstBatch"] |> Enum.map(&Map.fetch!(&1, "name"))
+  end
```
Comment (michalmuskala) [inline lib/mongo_ecto.ex]: `listCollections` is available since 3.0, we should use it since then. Let's only check the version is `> 3` - this should be enough and prepare us for next versions.

File: lib/mongo_ecto.ex [hunk 5]
```diff
@@ -758,7 +758,13 @@ defmodule Mongo.Ecto do
   @list_collections_query ["$and": [[name: ["$not": special_regex]],
                                     [name: ["$not": migration_regex]]]]
 
-  defp list_collections(repo, opts) do
+  def list_collections(_ = "3.2", repo, opts) do
+    colls = command(repo, %{"listCollections": 1}, opts)
+
+    colls["cursor"]["firstBatch"] |> Enum.map(&Map.fetch!(&1, "name"))
```
Comment (michalmuskala) [inline lib/mongo_ecto.ex]: Please don't use pipe with just one element. 

We also need to exclude the `schema_migrations` table.

File: lib/mongo_ecto/connection.ex [hunk 6]
```diff
@@ -124,22 +124,29 @@ defmodule Mongo.Ecto.Connection do
...
   end
 
-  defp constraint(msg) do
-    case String.split(msg, [".$", " dup "]) do
+  def constraint(msg) do
+    matchf = if String.contains?(msg, ".$") do
+      fn(msg) -> String.split(msg, [".$", " dup "]) end
+    else
+      fn(msg) -> String.split(msg, ["index: ", " dup "]) end
+    end
```
Comment (michalmuskala) [inline lib/mongo_ecto/connection.ex]: Wouldn't doing `String.split(msg, [".$", "index: ", " dup "])` work? If `$.` is not in the string we won't split on it.

File: lib/mongo_ecto/connection.ex (hunk 2 above)
Comment (kwojtaszek) [inline lib/mongo_ecto/connection.ex]: Do we really need to acquire stacktrace when we're using it only in this node? It increases readability or maybe it's pattern that should we use?

Comment (ericmj) [inline lib/mongo_ecto/connection.ex]: Other code can change the stacktrace which will cause very confusing error messages. Because of this you should always capture the stacktrace as early as possible.

Comment (kwojtaszek) [inline lib/mongo_ecto/connection.ex]: Cool, haven't thought about that, rolled back

File: lib/mongo_ecto.ex [hunk 7]
```diff
@@ -752,13 +752,27 @@ defmodule Mongo.Ecto do
...
                                     [name: ["$not": migration_regex]]]]
 
-  defp list_collections(repo, opts) do
+  def list_collections(repo, opts \\ []) do
+    ver = db_version(repo)
+
+    list_collections(ver, repo, opts)
+  end
+
+  defp list_collections(_ = 3, repo, opts) do
```
Comment (michalmuskala) [inline lib/mongo_ecto.ex]: Let's change it to:

```elixir
defp list_collections(version, repo, opts) when version >= 3 do
```

**Comment by michalmuskala** [comment]
This is great!

I've added one tiny comment and we're good to go. Could you squash the commits?

**Comment by michalmuskala** [comment]
Thank you! :heart: 

I'll release a new version with this today.
